File: test/clojure/bench_scaled/sieve.clj
;; Scaled mirror of test/jacl/bench_scaled/sieve.jacl — primes < 60000 = 6057.
;; Doubly-nested integer loop (trial division). loop/recur is *the* Clojure
;; looping form — this is the idiomatic naive version, matching the Python/JS mirrors.
(ns bench.sieve)

(defn run []
  (let [n 60000]
    (loop [k 2, cnt 0]
      (if (< k n)
        (let [is-p (loop [i 2]
                     (cond (> (* i i) k)     true
                           (zero? (rem k i)) false
                           :else             (recur (inc i))))]
          (recur (inc k) (if is-p (inc cnt) cnt)))
        cnt))))

File: test/lua/bench_scaled/sieve.lua
-- Scaled mirror of test/jacl/bench_scaled/sieve.jacl — primes < 60000 = 6057.
local function run()
  local n, count, k = 60000, 0, 2
  while k < n do
    local is_p, i = 1, 2
    while i * i <= k do
      if k % i == 0 then is_p = 0 end
      i = i + 1
    end
    if is_p == 1 then count = count + 1 end
    k = k + 1
  end
  return count
end
return { run = run }
